openedx/core/djangoapps/dark_lang/middleware.py

```diff
@@ -118,13 +118,15 @@ def _clean_accept_headers(self, request):
 
     def _activate_preview_language(self, request):
         """
-        Check the user's dark language setting in the session and apply it
+        Check the user's language setting in the session and apply it.
+
+        Neglect the dark language preview feature in the language settings check,
+        since we need to rely solely on the server lang configs as per SKILLONOMY-236.
         """
         auth_user = request.user.is_authenticated()
-        preview_lang = None
-        if auth_user:
-            # Get the request user's dark lang preference
-            preview_lang = get_user_preference(request.user, DARK_LANGUAGE_KEY)
+        preview_lang = get_user_preference(
+            request.user, LANGUAGE_KEY
+        ) or settings.LANGUAGE_CODE if auth_user else settings.LANGUAGE_CODE
```

Comment:
```if auth_user else settings.LANGUAGE_CODE``` is unnecessary.

Reply:
if anonymous, we don't need to check user preferences. 

`get_user_preference` seems to be crafted for authenticated users

Reply:
I'm merging)

Reply:
Fixed with #1988